engines/hypno/scene.cpp

```diff
@@ -0,0 +1,371 @@
+	for (Filenames::iterator it = trans.intros.begin(); it != trans.intros.end(); ++it) {
+		MVideo v(*it, Common::Point(0, 0), false, true, false);
+		runIntro(v);
+	}
+
+	if (!trans.frameImage.empty()) {
+		debugC(1, kHypnoDebugScene, "Rendering %s frame in transaction", trans.frameImage.c_str());
+		Graphics::Surface *frame = decodeFrame(trans.frameImage, trans.frameNumber);
+		Graphics::Surface *sframe = frame->scale(_screenW, _screenH);
+		drawImage(*sframe, 0, 0, false);
+		drawScreen();
+		frame->free();
+		delete frame;
+		sframe->free();
+		delete sframe;
+		Common::String *ptr = new Common::String(trans.level);
+		assert(installTimer(2 * 1000000, ptr));
+	} else
+		_nextLevel = trans.level;
+}
+
+
+void HypnoEngine::runScene(Scene &scene) {
+	_refreshConversation = false;
+	_conversation.clear();
+	Common::Event event;
+	Common::Point mousePos;
+	Common::List<uint32> videosToRemove;
+
+	stack.clear();
+	_nextHotsToAdd = &scene.hots;
+	defaultCursor();
+
+	while (!shouldQuit() && _nextLevel.empty()) {
+
+		while (g_system->getEventManager()->pollEvent(event)) {
+			mousePos = g_system->getEventManager()->getMousePos();
+			// Events
+			switch (event.type) {
+			case Common::EVENT_KEYDOWN:
+				if (event.kbd.keycode == Common::KEYCODE_ESCAPE) {
+					for (Videos::iterator it = _videosPlaying.begin(); it != _videosPlaying.end(); ++it) {
+						if (it->decoder)
+							skipVideo(*it);
+					}
+					_videosPlaying.clear();
+
+					if (!stack.empty()) {
+						runMenu(*stack.back());
+						drawScreen();
+					}
+				}
+
+				break;
+
+			case Common::EVENT_QUIT:
+			case Common::EVENT_RETURN_TO_LAUNCHER:
+				break;
+
+			case Common::EVENT_RBUTTONDOWN:
+				if (stack.empty())
+					break;
+				if (!_conversation.empty()) {
+					rightClickedConversation(mousePos);
+					break;
+				}
+				break;
+
+			case Common::EVENT_LBUTTONDOWN:
+				if (stack.empty())
+					break;
+				if (!_conversation.empty()) {
+					leftClickedConversation(mousePos);
+					break;
+				}
+				if (!_nextHotsToAdd && !_nextHotsToRemove /*&& _videosPlaying.empty()*/)
+					clickedHotspot(mousePos);
+				break;
+
+			case Common::EVENT_MOUSEMOVE:
+				// Reset cursor to default
+				//changeCursor("default");
+				// The following functions will return true
+				// if the cursor is changed
+				if (stack.empty() || !_conversation.empty() || !_videosPlaying.empty())
+					break;
+
+				if (!hoverHotspot(mousePos))
+					defaultCursor();
+				break;
+
+			default:
+				break;
+			}
+		}
+
+		if (_refreshConversation && !_conversation.empty() && _nextSequentialVideoToPlay.empty()) {
+			showConversation();
+			drawScreen();
+			_refreshConversation = false;
+			_videosPlaying.clear();
+		}
+
+		// Movies
+		if (!_nextSequentialVideoToPlay.empty() && _videosPlaying.empty()) {
+			playVideo(*_nextSequentialVideoToPlay.begin());
+			_videosPlaying.push_back(*_nextSequentialVideoToPlay.begin());
+			_nextSequentialVideoToPlay.remove_at(0);
+		}
+		uint32 i = 0;
```

Comment:
Looks like `i` is not used at all, only incremented.

Reply:
`i` is pushed into `videosToRemove`.